src/deepsparse/utils/onnx.py

```diff
@@ -303,7 +304,7 @@ def override_onnx_input_shapes(
             "Overwriting in-place the input shapes of the model " f"at {onnx_filepath}"
         )
         onnx.save(model, onnx_filepath)
-        return onnx_filepath
+        yield onnx_filepath
```

Comment:
Should the `return` on line 307/308 also be changed to `yield`? Given that nothing happens after the `yield`, is the change necessary?

Reply:
You’re right. Given that, though, the `return` in the `else` clause should definitely be changed to `yield`.

Reply:
Oh, actually this returns a function that in itself yields :)

Reply:
You may be right @dbarbuzzi. Just realized it writing the unit tests.